OpenSim/Simulation/Wrap/WrapCylinder.cpp

```diff
@@ -44,11 +44,28 @@ using SimTK::Vec3;
 using SimTK::UnitVec3;
 
 static const char* wrapTypeName = "cylinder";
-static Vec3 p0(0.0, 0.0, -1.0);
-static Vec3 dn(0.0, 0.0, 1.0);
+static const Vec3 g_CylinderBottom(0.0, 0.0, -1.0);
+static const UnitVec3 g_CylinderDirection(0.0, 0.0, 1.0);
 #define MAX_ITERATIONS    100
 #define TANGENCY_THRESHOLD (0.1 * SimTK_DEGREE_TO_RADIAN) // find tangency to within 1 degree
 
+// helper function for calling the wrapmath version
+static Vec3 GetClosestPointOnLineToPoint(const Vec3& pt,
+                                         const Vec3& linePt,
+                                         const UnitVec3& lineDir)
+{
+    Vec3 rv;
+    double t;
+    WrapMath::GetClosestPointOnLineToPoint(pt, linePt, lineDir, rv, t);
+    return rv;
+}
+
+static double CalcWrapAngle(const UnitVec3& a, const UnitVec3& b, bool reverse)
+{
+    double ang = acos(SimTK::dot(a, b));
+    return !reverse ? ang : -(2.0*SimTK_PI - ang);
```

Comment:
I think I see the logic but `return reverse ? (ang - 2.0*SimTK_PI) : ang;` might be easier to read. Also not sure what `reverse` is used for… if it comes from the old line 644
```
if (far_side_wrap)
    theta = 2.0 * SimTK_PI - theta;
```
then why is it now `ang - 2.0*SimTK_PI` rather than `2.0*SimTK_PI - ang`?

Reply:
For this one, it's because the original code was using a variable called `sense` to flip the magnitude of the angle so, in effect, it was being double-flipped by the time it reached the rotation step.

The reason I reversed the if was a silly premature optimization: most compilers have a concept of a "happy" path on branching code (assumed to be the true branch if the compiler can't guess the weighting). The happy branch usually does the comparison and then just keeps moving through the code, the unhappy branch requires a jump (this is why there's the concept of "unlikely" annotations in compilers).

It's not an applicable optimization here because this isn't a short method where skipping a jump and immediately returning would have advantages. I'll reverse it :)

Reply:
Thanks for the explanation. I wasn't aware of [the likely/unlikely attributes](https://en.cppreference.com/w/cpp/language/attributes/likely).